addons/sale_loyalty/models/sale_order.py

```diff
@@ -666,6 +666,12 @@ def _get_claimable_rewards(self, forced_coupons=None):
                 # If the total is 0 again without the payment reward it will be removed.
                 if reward.reward_type == 'discount' and total_is_zero and (not has_payment_reward or reward.program_id.is_payment_program):
                     continue
+                search_domain = expression.AND([
+                    reward._get_active_products_domain(),
+                    [('id', "=", reward.id)]
+                ])
+                if reward.search_count(search_domain) == 0:
```

Comment:
```suggestion
                if not reward.filtered_domain(reward._get_active_products_domain()):
```
You can probably rely on `filtered_domain` to avoid making a query to the database.